helios-client/src/main/java/com/spotify/helios/client/Response.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2015 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.client;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+class Response {
+
+  private final String method;
+  private final URI uri;
+  private final int status;
+  private final byte[] payload;
+  private final Map<String, List<String>> headers;
+
+  public Response(final String method, final URI uri, final int status, final byte[] payload,
+                  final Map<String, List<String>> headers) {
+    this.method = method;
+    this.uri = uri;
+    this.status = status;
+    this.payload = payload;
+    this.headers = headers;
+  }
+
+  @Override
+  public String toString() {
+    return "Response{" +
+           "method='" + method + '\'' +
+           ", uri=" + uri +
+           ", status=" + status +
+           ", payload=" + Arrays.toString(payload) +
```

Comment:
If we are always dealing with character-based payloads (and maybe we aren't) then logging the decoded version of the payload in a toString() would probably be more useful than the raw byte array